datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -89,10 +110,22 @@ impl CaseExpr {
         if when_then_expr.is_empty() {
             exec_err!("There must be at least one WHEN clause")
         } else {
+            let eval_method = if expr.is_some() {
+                EvalMethod::CaseWithExpression
+            } else if when_then_expr.len() == 1
+                && when_then_expr[0].1.as_any().is::<Column>()
+                && else_expr.is_none()
```

Comment:
If `else` value is a null literal, is `else_expr` still none? Or is it a null literal?

Reply:
From SQL planning, it will be `None`, but it makes sense to add code here to normalize this for other use cases (other query engines that delegate to DataFusion). I have added this.